src/indexer/index.ts

```diff
@@ -106,7 +106,12 @@ export async function indexEvents(
             })
             .transacting(trx)
         })
-        await setLastBlock(key, toBlock).transacting(trx)
+        if (events.length > 0) {
```

Comment:
This stills like it's worth a comment.  It's not obvious why this is required.

Reply:
Also, could this theoretically cause an infinite loop?  E.g., a span of blocks w/o events that's batchSize long.